File: client/lib/models/analysis_message.dart
enum MessageRole { user, assistant, system }

class AnalysisMessage {
  final String id;
  final String? dreamId;
  final String content;
  final MessageRole role;
  final DateTime createdAt;

  AnalysisMessage({
    required this.id,
    this.dreamId,
    required this.content,
    required this.role,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory AnalysisMessage.fromJson(Map<String, dynamic> json) {
    return AnalysisMessage(
      id: json['id'] as String,
      dreamId: json['dream_id'] as String?,
      content: json['content'] as String,
      role: MessageRole.values.firstWhere(
        (r) => r.name == json['role'],
        orElse: () => MessageRole.assistant,
      ),
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'dream_id': dreamId,
      'content': content,
      'role': role.name,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

File: backend/models/analysis_message.py
"""Модель сообщения анализа (чат по снам)"""

import uuid
import enum
from datetime import datetime
from sqlalchemy import String, Text, DateTime, ForeignKey, Index
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship
from database import Base


class MessageRole(str, enum.Enum):
    """Роли сообщений"""
    USER = "user"
    ASSISTANT = "assistant"
    SYSTEM = "system"


class AnalysisMessage(Base):
    """Сообщение в чате анализа снов"""

    __tablename__ = "analysis_messages"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4
    )
    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False
    )
    dream_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("dreams.id", ondelete="CASCADE"),
        nullable=True
    )
    role: Mapped[str] = mapped_column(
        String(16),
        nullable=False
    )
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=datetime.utcnow,
        nullable=False
    )

    # Связи
    user: Mapped["User"] = relationship("User", back_populates="analysis_messages")
    dream: Mapped["Dream"] = relationship("Dream", back_populates="messages")

    __table_args__ = (
        Index("ix_analysis_messages_user_created", "user_id", "created_at"),
        Index("ix_analysis_messages_user_dream_created", "user_id", "dream_id", "created_at"),
    )

    def __repr__(self) -> str:
        return f"<AnalysisMessage(id={self.id}, role={self.role}, dream_id={self.dream_id})>"
